components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -93,6 +86,46 @@ void Nudging::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
     m_num_src_levs = scorpio::get_dimlen(m_static_vertical_pressure_file,"lev");
     scorpio::eam_pio_closefile(m_static_vertical_pressure_file);
   }
+
+  /* Check for consistency between nudging files, map file, and remapper */
+
+  // Number of columns globally
+  auto m_num_cols_global = m_grid->get_num_global_dofs(); 
+
+  // Get the information from the first nudging data file
+  scorpio::register_file(m_datafiles[0],scorpio::Read);
+  int num_cols_src = scorpio::get_dimlen(m_datafiles[0],"ncol");
+  scorpio::eam_pio_closefile(m_datafiles[0]);
+
+  if (num_cols_src != m_num_cols_global) {
+    // If differing cols, check if remap file is provided
+    EKAT_REQUIRE_MSG(m_refine_remap_file != "no-file-given",
+                     "Error! Nudging::set_grids - the number of columns in the nudging data file "
+                     << std::to_string(num_cols_src) << " does not match the number of columns in the "
+                     << "model grid " << std::to_string(m_num_cols_global) << ".  Please check the "
+                     << "nudging data file and/or the model grid.");
+    // If remap file is provided, check if it is consistent with the nudging data file
+    // First get the data from the mapfile
+    scorpio::register_file(m_refine_remap_file,scorpio::Read);
+    int num_cols_remap_a = scorpio::get_dimlen(m_refine_remap_file,"n_a");
+    int num_cols_remap_b = scorpio::get_dimlen(m_refine_remap_file,"n_b");
+    scorpio::eam_pio_closefile(m_refine_remap_file);
+    // Then, check if n_a (source) and n_b (target) are consistent
+    EKAT_REQUIRE_MSG(num_cols_remap_a == num_cols_src,
+                     "Error! Nudging::set_grids - the number of columns in the nudging data file "
+                     << std::to_string(num_cols_src) << " does not match the number of columns in the "
+                     << "mapfile " << std::to_string(num_cols_remap_a) << ".  Please check the "
+                     << "nudging data file and/or the mapfile.");
+    EKAT_REQUIRE_MSG(num_cols_remap_b == m_num_cols_global,
+                      "Error! Nudging::set_grids - the number of columns in the model grid "
+                      << std::to_string(m_num_cols_global) << " does not match the number of columns in the "
+                      << "mapfile " << std::to_string(num_cols_remap_b) << ".  Please check the "
+                      << "model grid and/or the mapfile.");
+    // If we get here, we are good to go!
+    m_refine_remap = true;
+  } else {
+    m_refine_remap = false;
```

Comment:
I would still add a check on whether a map file was provided. It may be that the user _wanted_ remap, and added the map file, but forgot to change the nudging input file. You may just issue a warning in this case, rather than error out...